src/gowrapper/ckks/keygen.go

```diff
@@ -184,6 +198,75 @@ func lattigo_genRotationKeysForRotations(keygenHandle Handle5, skHandle Handle5,
 	return marshal.CrossLangObjMap.Add(unsafe.Pointer(rotKeys))
 }
 
+//export lattigo_getRotationKey
+func lattigo_getRotationKey(paramHandle, rotKeysHandle Handle5, rotationStep int) Handle5 {
+	param := getStoredParameters(paramHandle)
+	rotKeys := getStoredRotationKeys(rotKeysHandle)
+	rotationKey := rotKeys.Keys[param.GaloisElementForColumnRotationBy(rotationStep)]
+	return marshal.CrossLangObjMap.Add(unsafe.Pointer(rotationKey))
+
+}
+
+//export lattigo_rotationKeyExist
+func lattigo_rotationKeyExist(paramHandle, rotKeysHandle Handle5, rotationStep int) int {
+	param := getStoredParameters(paramHandle)
+	rotKeys := getStoredRotationKeys(rotKeysHandle)
+	_, exist := rotKeys.Keys[param.GaloisElementForColumnRotationBy(rotationStep)]
+	if exist {
+		return 1
+	} else {
+		return 0
+	}
+}
+
+//export lattigo_setRotationKey
+func lattigo_setRotationKey(paramHandle, rotKeysHandle, rotKeyHandle Handle5, rootStep int) {
+	param := getStoredParameters(paramHandle)
+	rotKeys := getStoredRotationKeys(rotKeysHandle)
+	rotKey := getStoredRotationKey(rotKeyHandle)
+	rotKeys.Keys[param.GaloisElementForColumnRotationBy(int(rootStep))] = rotKey
+
+}
+
+//export lattigo_copyNewRotationKey
+func lattigo_copyNewRotationKey(rotKeyHandle Handle5) Handle5 {
+	rotKey := getStoredRotationKey(rotKeyHandle)
+	return marshal.CrossLangObjMap.Add(unsafe.Pointer(rotKey.CopyNew()))
+}
+
+//export lattigo_numOfDecomp
+func lattigo_numOfDecomp(rotKeyHandle Handle5) uint64 {
+	rotKey := getStoredRotationKey(rotKeyHandle)
+	return uint64(len(rotKey.Value) - 1)
+}
+
+//export lattigo_galoisElementForColumnRotationBy
+func lattigo_galoisElementForColumnRotationBy(paramHandle Handle5, rotationStep uint64) uint64 {
+	param := getStoredParameters(paramHandle)
+	return uint64(param.GaloisElementForColumnRotationBy(int(rotationStep)))
+}
+
+//export lattigo_rotationKeyIsCorrect
+func lattigo_rotationKeyIsCorrect(rotKeyHandle Handle5, galEl uint64, skHandle Handle5, paramHandle Handle5, log2Bound uint64) uint64 {
+	rotKey := getStoredRotationKey(rotKeyHandle)
+	sk := getStoredSecretKey(skHandle)
+	param := getStoredParameters(paramHandle)
+	rotKey.GadgetCiphertext.CopyNew()
+	rotKey.CopyNew()
+	isCorrect := rlwe.RotationKeyIsCorrect(rotKey.CopyNew(), galEl, sk.CopyNew(), param.Parameters, int(log2Bound))
+	if isCorrect {
+		return uint64(1)
+	} else {
+		return uint64(0)
+	}
+}
+
+//export lattigo_ciphertextQP
+func lattigo_ciphertextQP(rotKeyHandle Handle5, i, j uint64) Handle5 {
```

Comment:
the name `lattigo_ciphertextQP` is confusing. maybe something like `lattigo_rotKeyCiphertextQP`